### ISSUE TITLE
drop executes property chain: realizes o concretizes

### ISSUE THREAD

**Issue by wdduncan** [OBOFoundry/COB#217]
The `executes` relation includes the property chain `realizes o concretizes`. I realize the property chain is included probably b/c this is how it is defined in STATO.

However, I am skeptical that the property chain does any real work for anyone, and I don't think the notion of realizing concretizations makes much sense. I'll skip my rant about this. I would like to use this relation, but can't bring myself to do so with this property chain included.

**Comment by wdduncan** [comment]
Another comment:
The label `executes` is too ambiguous. Machines/people (in general agents) execute plan specifications. The current `executes` relation doesn't permit this.  
I think it is better to disambiguate how a plan is executed with relations such as `agent executes` and `executed during` (for relation specifications to processes).

**Comment by zhengj2007** [comment]
I am OK with label 'executes'. But the relation should specify the domain and range. I don't prefer the label 'agent executes' since 'agent' is not defined in the ontology.

**Comment by ddooley** [comment]
Planned process "has performer" [operator/agent/robot/device] is current relation one can use to attach to an agents. True, its "('Homo sapiens' or organization or device) and ('has role' some 'investigation agent role')" axiom doesn't quite fit general purpose with its "investigation role". 

Currently planned process "executes" (COB) plan specification.  

I'll be a touch facetious in saying a relation between a performer and a plan specification could be "follows instructions of"

**Comment by wdduncan** [comment]
@zhengj2007  
> I don't prefer the label 'agent executes' since 'agent' is not defined in the ontology.
Fair point. However `agent role` is defined, so an `agent` would be an entity that bears an `agent role`.

> I am OK with label 'executes'. But the relation should specify the domain and range. 
Since `executes` has the property chain `realizes o concretizes`, the domain logically must be a process; only a process can realize things. Don't machines also "execute" plans?

**Comment by zhengj2007** [comment]
> However `agent role` is defined, so an `agent` would be an entity that bears an `agent role`

I found the 'agent role' is defined in two ontologies, [OBI_0000202](http://purl.obolibrary.org/obo/OBI_0000202) 'investigation agent role' and [SEPIO_0000048](http://purl.obolibrary.org/obo/SEPIO_0000048) 'contributor role'. The OBI one might be too specific. If we'd like to emphasize who is the performer of the process, I am fine to relabel the relation as 'agent executes' but need to define what 'agent' is.

The range of the relation should be GDC, right?

**Comment by wdduncan** [comment]
@zhengj2007 COB defines agent role as:
```
A role realized by a participant in a process such that the participant causes the process.
```
This seems okay to me. 

However, as @ddooley mentioned [above](https://github.com/OBOFoundry/COB/issues/217#issuecomment-1332462156) OBI now has a `has performer` relation (though I'm not sure if has been finalized yet). It seems to me that this would work fine for relating a `material entity` to a `process`.

**Comment by wdduncan** [comment]
@zhengj2007 @ddooley This issue has gotten far away from why I opened it, which as to remove the property chain `realizes o concretizes` from `executes`.

**Comment by zhengj2007** [comment]
I think 'executes' was proposed for simplifying the relation 'realizes o concretizes' used when associate a process with an ICE. So, I'd like to keep the property chain.

**Comment by ddooley** [comment]
Maybe a test question is whether "realizes o concretizes" has ever been used in OBO community in a non-anonymous way.  i.e. has someone ever said x realizes y, and y concretizes z, where y is a class or an instance with other stuff about it? If not, then it does seem a dispensable compexity.

**Comment by zhengj2007** [comment]
@wdduncan Thanks! I will check the discussion on bfo-discuss. 

I'd like to keep 'concretization of'. I think it is some mysterious part or black box that happens to convert an ICE generally in a person's brain as realizable entity and than realize in a process. 
Discussed with Barry several years ago. We agreed that there are two kinds of concretization, one is concretize an ICE as a realizable entity and realized in a process. Another kind of concretization is concretize an ICE as a quality that inheres in a material entity. So, I'd prefer to have more specific concretization relations rather than one relation between GDC and SDC, which are one relation is ICE <-> realizable entity, and the other is ICE <-> quality.

**Comment by wdduncan** [comment]
@zhengj2007 I am quite familiar with the notion of ICEs being concretized as either a quality or realizable. Concretizations as qualities makes an intuitive appeal that we (and machines) understand/process information by inputing (through our senses) patterns of qualities. Concretizations as realizables makes no such intuitive appeal (at least to me). You just have a vague notion that an ICE is somehow mysteriously transformed into a realizable that can only be accessed via a process

I am not denying that we (and machines) "execute" plans. We do. But (as noted above), I think we are able to carry out plan specifications b/c we have cognitive capabilities that permit us to read, evaluate, and carry out such specifications. This ability is inherent in us, and not in some realizable concretization.

If you already think realizable concretizations make sense, I doubt I can persuade you otherwise on Github. I am only requesting that the property chain be dropped b/c it is contentious. A less ambiguous label would be nice too.

**Comment by alanruttenberg** [comment]
According to your intuition, how are realizable entities created?

**Comment by wdduncan** [comment]
@alanruttenberg Hah! Not going down that rabbit hole :)

**Comment by alanruttenberg** [comment]
Well, absent an articulation of how your intuition works we just have a vague notion there might be a problem and it's a mystery (at least to me) how you might justify it.  ;-)

**Comment by wdduncan** [comment]
I'm not advocating the notion of realizable concretizations. So, I don't follow why I have to give some account of how realizable are created.

I've already gave a thumbnail account of what I think: We have cognitive processes that permit use to read, evaluate, and carry out instructions. This why we can do things such as change our minds about plans. No realizable concretizations are needed in this account.

You can weave some complex story about how realizable concretizations permit too. But why do so if you don't need to? If the burden is on anyone to explain (or discuss intuitions) how realizable concretizations are created, I think it is on those who hold to realizable concretizations.

Again, I'm not expecting anyone who advocates realizable concretizations to be convinced to change their minds here. I am on wanting the property chain to not be included in COB. This allows us who disagree with the property chain to use the executes relation.

**Comment by wdduncan** [comment]
Also
> Well, absent an articulation of how your intuition works we just have a vague notion there might be a problem and it's a mystery (at least to me) how you might justify it. ;-)

There is already a long discussion about this on bfo-discuss. Do we need to recapitulate that here?

**Comment by wdduncan** [comment]
Thanks @zhengj2007 I opened the ticket b/c I think the property chain should be dropped.

**Comment by hoganwr** [comment]
+1 for dropping the property chain.

**Comment by bpeters42** [comment]
+1 for keeping the property chain. I have been meaning to weigh in for a while, and hope to do so over the week.

**Comment by jamesaoverton** [comment]
I think we should keep the property chain. There certainly is not consensus for dropping it.

**Comment by ddooley** [comment]
I've been trying to understand this topic.  Setting aside discussion of those information content entities that can be concretized as qualities, is the issue that an ICE like a protocol or programming code, **by itself** is not concretized into a realizable?!  Is it that such an ICE is just a specified input to a process which is realizing some biological or mechanical or electronic function?  Such a function does depend on (it "uses") an ICE construct (protocol or code, say), but I don't think we have an RO relation linking the two.

**Comment by wdduncan** [comment]
@ddooley   
>  is the issue that an ICE like a protocol or programming code, by itself is not concretized into a realizable?

Yes. This is the issue. I don't think this makes sense. On this issue thread, it looks like some agree with me and some don't.

**Comment by alanruttenberg** [comment]
Haven't we beaten this to death? It's in OBI and has been for 10 years. It's not going anywhere and while it doesn't make sense to Bill, it makes sense to a bunch other of us.

**Comment by alanruttenberg** [comment]
Here's a fun one that I hadn't thought of in the past. What is the ontological story around subliminal advertising?

**Comment by wdduncan** [comment]
> It's in OBI and has been for 10 years. It's not going anywhere and while it doesn't make sense to Bill, it makes sense to a bunch other of us.  

I was mainly replying to Damion. However, just because it is OBI doesn't necessarily mean it needs to be in COB. Like I said it not everyone agrees with this.

> What is the ontological story around subliminal advertising?  
I don't know. Is there some kind of magical concretized disposition at work?

**Comment by alanruttenberg** [comment]
Let's try to unpack this. You use the term "Magical". How are we to understand what "magical" means? Here's some possible interpretations 
- Has no physical basis
- Has no physical precedent
- Has no physical direct cause
Provide another if you think it helps your argument. Now obviously I don't think anything doesn't have, ultimately, some basis in something actual happening in the world. So no, there is no "magical" process by which a concretization that is a disposition comes to exist that is any more magical than a process by which a concretization that is a quality comes to exist.

What do we know? We know there's a message, and information content entity,  many concretizations of the subliminal message. We know that at some point there is this disposition. In subliminal advertising I think it's reasonable to believe that there isn't a memory of the message content in the usual sense in which we recite back a poem that we have learned.  There's a reasonably proximate cause for the existence of the disposition, namely the information presentation. The proximity seems similar to the proximity of an information presentation and a resulting quality.

Now what is it that you consider magical?
- That there are dispositions at all?
- That there are specifically dependent specifically continuants that are specifically dependent on the brain? 
- That there are dispositions that are specifically dependent on the brain?
- That there are not processes (let's call them copy processes) that have as input information entities (or their concretizations)  and outputs things which are concretizations of them.
- That there are specifically dependent continuants created as the result of processes external to the brain? 
- That communication processes are not the sort of processes that result in the creation of such specifically dependent continuants?
- That there are but those only result in quality concretizations?

ALL of these are the sort of things someone might call "magical". You are deciding there's a boundary that's clear between the magical and unmagical things that happen. There must be some basis for this belief. What is it?

Here's what I presuppose and how I get there. I believe that the brain can bear specifically dependent continuants. I believe there are processes that originate outside the brain that result in the creation of them. I believe those specifically dependent continuants can be in some cases realizable entities. I believe that among those cases those specifically dependent continuants there can be concretizations. 

In order to believe that I'm obviously drawing a certain kind of circle around the process that originates externally and which ends in the creation of the specifically dependent continuant that happens to be a concretization.  We always have to draw a circle around processes - all processes are fiat in some sense. And there are things I don't understand along the way whether it is a quality or disposition that is in the end created. I suspect there are many steps between perception and the resulting SDCs. None of us know enough to perfectly characterize what all those processes are.

I see no reason to believe that there is a sharp boundary between processes that result in quality SDCs and those that result in realizable SDCs. I see no reason to believe that there is sharp boundary between, in delimiting such processes, those processes which initiate with presentation of information and those that do not.

Moreover there is a problem which we have to acknowledge that there isn't an obvious way to distinguish which things are at the bottom, or are most fundamental, between qualities and dispositions. Do all qualities have a dispositional basis, or is it the other way around? I don't think that question can be answered. There are good arguments for both.

I don't think there's any reason to believe that memories, or mental states, or whatever the mental functioning ontology classifies as qualities are fundamentally qualities or dispositions. I think we land up having to believe it's turtles all the way down. We can alternate between quality bases and dispositional bases going as far down in physics as you would like.

Obviously some of the things I believe, you do not. Which ones and why?

Saying that concretizations can only be quality SDCs but not realizable seems completely arbitrary to me. I know no argument that would justify that conclusion. So I don't rule out concretizations as realizables and I don't see how you can. 

What I see in the case of subliminal advertising is a certain reasonable circumscription of a process that starts with an information entity and ends with a trace that is whatever dispositions are in the brain. I see no reason to think that those dispositions aren't concretizations.

To my mind you're not liking or believing in them or thinking they are magical is completely arbitrary when we consider all the things we believe must exist when we work within the framework of the Basic Formal Ontology.

**Comment by wdduncan** [comment]
So, the "magical" comment was snarky. Sorry that I hit a nerve.

I'm not going respond to respond to your points b/c I don't think there are any arguments that I can make that you (and others) would find convincing. Please don't take that remark personally. I like you very much :)

I'm working on manuscript that I hope to have submitted in a few months detailing my (and others) views about realizable concretizations.

**Comment by alanruttenberg** [comment]
> I'm not going respond to respond to your points b/c I don't think there are any arguments that I can make that you (and others) would find convincing. Please don't take that remark personally. I like you very much :)

In that case I move that we close this issue. I've laid out my reasoning. If you can't respond with a convincing counterargument argument, then this discussion is over. (However much I like you, which is a lot!)

Perhaps reopen the issue when your paper is published.

**Comment by wdduncan** [comment]
Here is a link to my paper that discusses what I don't like about realizing concretizations:  
https://jbiomedsem.biomedcentral.com/articles/10.1186/s13326-024-00315-0

You can also watch my presentation on [YouTube](https://www.youtube.com/watch?v=i4YRn8U4Hpc&t=104s).

My arguments will probably not convince everyone interested in this issue. But, at least it makes my position clear.

**Comment by wdduncan** [comment]
I've meaning to add to this thread for a while. Now that [information specifies](http://purl.obolibrary.org/obo/RO_0017002) and [specified by information](http://purl.obolibrary.org/obo/RO_0017010) have been added to RO, a reasonable way to relate plan specifications to planned processes is to define planned process as:
```
process and 'specified by information' some 'plan specification'
```
This avoids issues discussed before.

If you need to link the planned process to a realizable entity, this will/should work.  
Define plan as:
```
'realizable entity` and 'specified by information' some 'plan specification'
```
and define planned process as:
```
process and (realizes some plan)
```
